cstar/entrypoint/worker/worker.py

```
@@ -0,0 +1,479 @@
+        """A unique directory for this simulation run to write outputs."""
+        self._simulation: Simulation = ROMSSimulation.from_blueprint(
+            blueprint=self._blueprint_uri,
+            directory=self._output_dir,
+            start_date=request.start_date,
+            end_date=request.end_date,
+        )
+        """The simulation instance created from the blueprint."""
+        self._handler: ExecutionHandler | None = None
+        """The execution handler for the simulation."""
+        self._job_config = job_cfg
+        """Configuration for submitting jobs to an HPC."""
+
+    @staticmethod
+    def _get_unique_path(root_path: pathlib.Path) -> pathlib.Path:
+        """Create a unique path name to avoid collisions.
+
+        Parameters
+        ----------
+        root_path: pathlib.Path
+            The parent directory where the unique directory will be created.
+
+        Returns
+        -------
+        pathlib.Path
+            A unique path based on the current date and time.
+        """
+        current_time = datetime.now(timezone.utc)
+        return root_path / f"{current_time.strftime('%Y%m%d_%H%M%S')}"
+
+    def _prepare_file_system(self) -> None:
+        """Ensure fresh directories exist for the simulation outputs.
+
+        Removes any pre-existing directories and creates empty directories to avoid
+        collisions.
+        """
+        # a leftover .cstar.env cause empty repo/compilation errors; remove.
+        user_env_path = cstar_sysmgr.environment.user_env_path
+        if user_env_path.exists():
+            self.log.debug(f"Removing existing user env: {user_env_path}")
+            user_env_path.unlink()
+
+        # a leftover root_dir may have files in it, breaking download; remove.
+        if self._output_root.exists():
+            self.log.debug(f"Removing existing output dir: {self._output_root}")
+            shutil.rmtree(self._output_root)
```

Ok, per my previous comment, I'm scared now about someone trying to reuse a directory and maybe wiping away a bunch of previous data. I do see there's a difference between output_root and output_dir, though actually output_dir is the unique one, so won't this always delete previous sims even if we did make a unique dir? Maybe there's some move/copy stuff later. Regardless, I'd rather raise an error here than rmtree, just to be safe, until we understand usage patterns better. I really don't want to accidentally wipe someone's work.

---

Replaced the `rmtree` with raising an exception until we determine the uniqueness of output paths.